router-tests/mcp_hot_reload_test.go

```diff
@@ -0,0 +1,191 @@
+package integration
+
+import (
+	"encoding/json"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+)
+
+func TestMCPOperationHotReload(t *testing.T) {
+	// create a temp graphql file into mcp_operations
+	mcpOperationsDirectory := "./testdata/mcp_operations"
+	fileName := "getEmployeeNotes.graphql"
+	filePath := path.Join(mcpOperationsDirectory, fileName)
+
+	t.Parallel()
+
+	t.Run("List Updated User Operations On Addition and Removal", func(t *testing.T) {
+		testenv.Run(t, &testenv.Config{
+			MCP: config.MCPConfiguration{
+				Enabled: true,
+				HotReloadConfig: config.MCPOperationsHotReloadConfig{
+					Enabled:  true,
+					Interval: 5 * time.Second,
+				},
+			},
+		}, func(t *testing.T, xEnv *testenv.Environment) {
+
+			os.Remove(filePath)
```

Comment:
You should use a `t.TempDir` to modify files during tests, not the worktree, they get automatically cleaned up after the test finishes.